ga4gh/backend.py

```diff
@@ -192,6 +192,63 @@ def _getEnd(cls, variant):
         return variant.end
 
 
+class VariantAnnotationsIntervalIterator(IntervalIterator):
+    """
+    An interval iterator for annotations
+    """
+
+    def __init__(self, request, parentContainer):
+        super(VariantAnnotationsIntervalIterator, self).__init__(
+            request, parentContainer)
+        self.featureIds = set(self._request.feature_ids)
+        self.effects = set(self._request.effects)
+
+    def _search(self, start, end):
+        return self._parentContainer.getVariantAnnotations(
+            self._request.referenceName, start, end)
+
+    @classmethod
+    def _getStart(cls, annotation):
+        return annotation.variant.start
+
+    @classmethod
+    def _getEnd(cls, annotation):
+        return annotation.variant.end
+
+    def next(self):
+        while True:
+            ret = super(VariantAnnotationsIntervalIterator, self).next()
+            vann = ret[0]
+            if self.filterVariantAnnotation(vann):
+                return ret
+        return None
+
+    def filterVariantAnnotation(self, vann):
+        if not(self.featureIds or self.effects):
```

Comment:
I'm pretty critical of using implicit casting to boolean, as I think it's nearly always harder to read and can contain tricky exceptional cases that are hard to spot. PEP8 recommends against it also. It would be helpful to rephrase these conditionals in terms of explicit comparisons so it's clearer. For example, I would change the first to
```python
if len(self._featureIds) == 0 and len(self._effects) == 0:
    return True
```